level2/restaurants/adriana_restaurants.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+# class for Restaurants selling Items
+class Restaurant
+  attr_reader :name, :items
+
+  def initialize(name)
+    @name = name
+    @items = []
+  end
+
+  def add_items(*items)
+    @items << items
+    @items.flatten!
+  end
+
+  def remove_item(item_name)
+    @items.delete_if { |i| i.name == item_name }
+  end
+
+  def place_order(ordered_items, discount: 0)
+    return 'Double discount is not allowed' if invalid_discount?(ordered_items, discount)
+
+    order_price = 0
+    ordered_items.each do |ordered_item|
+      item = select_item(ordered_item[:name])
+
+      return 'Items not available' if item.nil? || !item.in_stock?
```

Comment:
its a good practice to not have negations in conditions when not necessary. like in this case you have `in_stock?` method that is used once and its being negated, so you could have just `out_of_stock?` instead - or both:

```
def in_stock?
  true
end

def out_of_stock?
  !in_stock?
end
```